addons/stock/tests/test_move.py

```diff
@@ -6524,3 +6524,41 @@ def test_set_quantity_done_with_rounding_issues(self):
             {'quantity': 149.97, 'quantity_product_uom': 5.29},
             {'quantity': 0.03, 'quantity_product_uom': 0},
         ])
+
+    def test_edit_lot_id_in_picking_form(self):
+        lot1 = self.env['stock.lot'].create({
+            'name': 'lot1',
+            'product_id': self.product_serial.id
+        })
+        lot2 = self.env['stock.lot'].create({
+            'product_id': self.product_serial.id
+        })
+        self.env['stock.quant']._update_available_quantity(self.product_serial, self.stock_location, 2)
+        self.env['stock.quant']._update_available_quantity(self.product_serial, self.stock_location, 1, lot_id=lot1)
+        self.env['stock.quant']._update_available_quantity(self.product_serial, self.stock_location, 1, lot_id=lot2)
+        picking = self.env['stock.picking'].create({
+            'location_id': self.stock_location.id,
+            'location_dest_id': self.customer_location.id,
+            'picking_type_id': self.picking_type_out.id,
+            'state': 'draft',
+        })
+        self.env['stock.move'].create({
+            'location_id': self.stock_location.id,
+            'location_dest_id': self.customer_location.id,
+            'product_id': self.product_serial.id,
+            'product_uom': self.uom_unit.id,
+            'product_uom_qty': 1.0,
+            'picking_id': picking.id,
+            'picking_type_id': self.picking_type_out.id,
+        })
+        picking.action_confirm()
+        with Form(picking) as picking_form:
+            with picking_form.move_ids.edit(0) as line_form:
+                line_form.lot_ids.remove(lot1.id)
+                line_form.lot_ids.add(lot2)
+            picking = picking_form.save()
+        self.assertEqual(len(picking.move_ids), 1)
```

Comment:
What is the purpose of checking the number of moves? Here it's certain that we only have one move.